drivers/cc110x/cc110x_communication.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup   drivers_cc110x
+ * @{
+ *
+ * @file
+ * @brief     Functions to communicate with the CC1100/CC1101 transceiver
+ *
+ * @author    Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ * @}
+ */
+
+#include <errno.h>
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "xtimer.h"
+#include "cc110x.h"
+#include "cc110x_constants.h"
+
+int cc110x_power_on(cc110x_t *dev)
+{
+    gpio_t cs = dev->params.cs;
+
+    if (gpio_init(cs, GPIO_OUT)) {
+        return -EIO;
+    }
+    gpio_clear(cs);
+    xtimer_usleep(150);
+    gpio_set(cs);
+    spi_init_cs(dev->params.spi, dev->params.cs);
+    return 0;
+}
+
+uint8_t cc110x_read(cc110x_t *dev, uint8_t addr, uint8_t *dest)
+{
+    uint8_t status;
+
+    addr |= CC110X_SINGLE_BYTE_READ;
+    spi_transfer_bytes(dev->params.spi, dev->params.cs, true,
+                       &addr, &status, 1);
+    spi_transfer_bytes(dev->params.spi, dev->params.cs, false, NULL, dest, 1);
+    return status;
+}
+
+uint8_t cc110x_read_workarround(cc110x_t *dev, uint8_t addr, uint8_t *dest)
```

Comment:
```suggestion
 uint8_t cc110x_read_workaround(cc110x_t *dev, uint8_t addr, uint8_t *dest)
```

Reply:
Fixed in 1ca319ba8fc188272081768c7712fd659dcef111